crates/cast/src/args.rs

```diff
@@ -694,11 +694,21 @@ pub async fn run_command(args: CastArgs) -> Result<()> {
             &mut std::io::stdout(),
         ),
         CastSubcommand::Logs(cmd) => cmd.run().await?,
-        CastSubcommand::DecodeTransaction { tx } => {
-            let tx = stdin::unwrap_line(tx)?;
-            let tx = SimpleCast::decode_raw_transaction(&tx)?;
```

Comment:
here we really just need to check if we can do

```rust
if let Ok(signer) = tx.recover_signer() {
   serde_json::to_string_pretty(&Recovered::new_uncheck(tx,signer))?
} else {
serde_json::to_string_pretty(&tx)?
} 
```

Reply:
Pushed this, do check it out :)